From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Mon, 19 Aug 2024 21:35:49 -0500
Subject: [PATCH] Fix space breaking macros

slibtool apparently cannot handle an extra space before the first argument. This breaks it completely if SLIBTOOL_FLAGS has no value since an extra space will be present in the variable.

To fix this use the strip function to remove whitespace.

---
 m4/slibtool.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/m4/slibtool.m4 b/m4/slibtool.m4
index bf1c4e1..f4c8bb4 100644
--- a/m4/slibtool.m4
+++ b/m4/slibtool.m4
@@ -525,7 +525,7 @@ _SLIBTOOL_ARG_WITH([sysroot],[absolute path to the target's sysroot],[slibtool_s
 # --------------------
 slibtool_set_flavor
 LIBTOOL='$(SLIBTOOL)'
-SLIBTOOL="${SLIBTOOL} \$(SLIBTOOL_FLAGS)"
+SLIBTOOL="\$(strip ${SLIBTOOL} \$(SLIBTOOL_FLAGS))"
 
 AC_SUBST([LIBTOOL])
 AC_SUBST([SLIBTOOL])
